onyo/lib/onyo.py

```diff
@@ -92,20 +92,22 @@ def clear_caches(self) -> None:
         self._gitfiles = None
         self._templates = None
 
-    def generate_commit_message(self, message: str = "", cmd: str = "",
+    def generate_commit_message(self, message: list[str] = [], cmd: str = "",
                                 keys: list[str] = [], destination: str = "") -> str:
         """
-        Returns either the commit message given to onyo by the user, or creates
-        a default commit message.
+        The commit message is generated based on arguments for `--message` if
+        given by the user, or a default commit message.
+        Adds a list of changed files to the body of commit messages.
         """
         message_header = ""
+        message_body = ""
         MAX_LEN = 80
+        assets = [Path(self.opdir, x).relative_to(self.root) for x in sorted(self.files_staged)]
 
-        # TODO: first construct body?
         if message:
-            message_header = message
+            message_header = "".join(message[0])
+            message_body = "\n".join(["".join(x) for x in message[1:]]) + "\n\n"
```

Comment:
`_n_join()` can be useful here.

```
message_body = self._n_join(message[1:]) 
```

The final `+ "\n\n"` should be done later, when it's assembled in context.